pop-cli-for-appchains/guides/benchmarking-a-parachain/benchmarking-pallets-and-extrinsics.md

```diff
@@ -0,0 +1,259 @@
+│
+◆  Select the runtime:
+│  ○ pop-runtime-testnet
+│  ● pop-runtime-devnet
+│  ○ pop-runtime-mainnet
+└
+```
+
+If the binary is missing, Pop CLI builds it with the appropriate `profile` and features. You can manually specify the runtime binary path using:
+
+```bash
+pop bench pallet --runtime=target/release/pop-runtime-devnet.wasm
+```
+
+By default, whenever benchmarking starts, runtime binary will be automatically built. You can provide a flag `--no-build` or `-n` to skip the build process if there is an existing runtime binary.
+
+```bash
+pop bench pallet --no-build
+```
+
+**Configure the genesis builder policy and preset**
+
+```bash
+◆  Select the genesis builder policy:
+│  ● none (Do not provide any genesis state.)
+│  ○ runtime
+└
+```
+
+Genesis builder policy defines the way to construct the [initial genesis state](https://docs.polkadot.com/develop/parachains/deployment/generate-chain-specs/). There are two options:
+
+- `none`: Do not provide any genesis state.
+- `runtime`: Use the runtime's genesis preset.
+
+To configure the genesis builder manually, you can use the `--genesis-builder` flag. For example:
+
+```bash
+pop bench pallet --runtime=target/release/pop-runtime-devnet.wasm --genesis-builder=runtime
+```
+
+The genesis preset is configured on the runtime via [`sp_genesis_builder`](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/) crates which contains a runtime-api to be implemented by runtimes, in order to express their genesis state.
+
+```bash
+◇  Select the genesis builder policy:
+│  runtime
+│
+◇  Found 2 genesis builder presets
+│
+◆  Select the genesis builder preset:
+│  ● development
+│  ○ local_testnet
+└
+```
+
+To configure the genesis builder preset, you can use the `--genesis-builder-preset` flag:
+
+```bash
+pop bench pallet --runtime=target/release/pop-runtime-devnet.wasm --genesis-builder=runtime --genesis-builder-preset=development
+```
+
+**Select pallets and extrinsics**
+
+You'll be prompted to benchmark all pallets or select a specific one. Pop CLI lists available pallets, allowing you to search by name and choose from the list.
+
+```bash
+◆  🔎 Search for a pallet to benchmark
+│
+│  ● cumulus_pallet_parachain_system
+│  ○ cumulus_pallet_xcmp_queue
+│  ○ frame_system
+│  ○ pallet_balances
+│  ○ pallet_collator_selection
+│  ○ pallet_message_queue
+│  ○ pallet_session
+│  ○ pallet_sudo
+│  ○ pallet_timestamp
+└
+```
+
+You can simply search for pallets by typing their name and selecting them from the list.
+```bash
+◆  🔎 Search for a pallet to benchmark
+│  pallet_time
+│  ● pallet_timestamp
+│  ○ pallet_session
+│  ○ pallet_sudo
+│  ○ pallet_message_queue
+│  ○ pallet_balances
+│  ○ pallet_collator_selection
+└
+```
+
+If a pallet is specified, the CLI will prompt you to select the extrinsics you want to benchmark within that pallet. You can select multiple extrinsics by pressing the spacebar. Same as searching for pallets, you can search for extrinsics by typing their name as well.
```

Comment:
We seem to use extrinsics and dispatchables interchangeably. I know the underlying tooling uses extrinsic, as you are technically submitting extrinsics (transactions) to measure their performance, but I think we should use 'dispatchables' or 'dispatchable functions' when referring to the functions by name. 'functions' would also be fine, as in pallet functions.